pkg/cache/task.go

```diff
@@ -533,6 +533,25 @@ func (task *Task) releaseAllocation() {
 // this reduces the scheduling overhead by blocking such
 // request away from the core scheduler.
 func (task *Task) sanityCheckBeforeScheduling() error {
+	// After version 1.7.0, we should reject the task whose pod is unbound and has conflicting metadata.
+	if !utils.PodAlreadyBound(task.pod) {
+		if err := utils.CheckAppIdInPod(task.pod); err != nil {
+			log.Log(log.ShimCacheTask).Warn("The task has conflicting metadata will be rejected after version 1.7.0.",
+				zap.String("appID", task.applicationID),
+				zap.String("podName", task.pod.Name),
+				zap.String("error", err.Error()))
+		}
+		if err := utils.CheckQueueNameInPod(task.pod); err != nil {
+			log.Log(log.ShimCacheTask).Warn("The task has conflicting metadata will be rejected after version 1.7.0.",
+				zap.String("appID", task.applicationID),
+				zap.String("podName", task.pod.Name),
+				zap.String("error", err.Error()))
+		}
+	}
+	return task.checkPodPVCs()
```

Comment:
In addition to the logged messages, we should also send pod events to inform the pod submitter of the potential issue.

Reply:
The pod events need to indicate which fields are inconsistent as well. For example:

Pod has inconsistent applciation metadata and may be rejected in a future YuniKorn release: label applicationID "foo" doesn't match annotation yunikorn.apache.org/app-id